extensions/exposition/source/HTTP/Server.ts

```diff
@@ -86,7 +86,7 @@ export class Server extends Connector {
 
       response
         .status(status)
-        .set(message.headers)
+        .set(Object.fromEntries(message.headers?.entries() ?? []))
```

Comment:
If there is no `headers`, this code will allocate memory for an empty array, then it will allocate memory for an empty object, created from that array, then this empty object will be iterated within `set`, most likely using another allocation for an empty array (key-value pairs). When these useless actions are finished, `GC` will wait his turn to block the process and clear all this mess.

This is hot code path, executed on each request. Please refactor it to something more efficient (like explicit check and iteration using `headers.forEach`).